tests/unit/moduleapi/propagate.tcl

```diff
@@ -21,22 +24,238 @@ tags "modules" {
+                    # This test is here only for sanity
+
+                    wait_for_condition 500 10 {
+                        [$replica dbsize] eq 1
+                    } else {
+                        fail "Not all keys have been evicted"
+                    }
+
+                    $master config set maxmemory 0
+                    $master config set maxmemory-policy noeviction
+                }
+
+                test {module propagation with notifications with eviction case 2} {
+                    $master flushall
+                    set repl [attach_to_replication_stream]
+
+                    $master set asdf1 1 EX 300
+                    $master set asdf2 2 EX 300
+                    $master set asdf3 3 EX 300
+
+                    # Please note we use volatile eviction to prevent the loop described in the test above.
+                    # "notifications" is not volatile so it always remains
+                    $master config set maxmemory-policy volatile-ttl
+                    $master config set maxmemory 1
+
+                    wait_for_condition 500 10 {
+                        [$replica dbsize] eq 1
+                    } else {
+                        fail "Not all keys have been evicted"
+                    }
+
+                    $master config set maxmemory 0
+                    $master config set maxmemory-policy noeviction
+
+                    $master set asdf4 4
+
+                    # Note whenever there's double notification: SET with EX issues two separate
+                    # notifications: one for "set" and one for "expire"
+                    assert_replication_stream $repl {
+                        {select *}
+                        {multi}
+                        {incr notifications}
+                        {incr notifications}
+                        {set asdf1 1 PXAT *}
+                        {exec}
+                        {multi}
+                        {incr notifications}
+                        {incr notifications}
+                        {set asdf2 2 PXAT *}
+                        {exec}
+                        {multi}
+                        {incr notifications}
+                        {incr notifications}
+                        {set asdf3 3 PXAT *}
+                        {exec}
+                        {incr notifications}
+                        {del asdf*}
+                        {incr notifications}
+                        {del asdf*}
+                        {incr notifications}
+                        {del asdf*}
+                        {multi}
+                        {incr notifications}
+                        {set asdf4 4}
+                        {exec}
+                    }
+                    close_replication_stream $repl
+                }
+
+                test {module propagation with timer and CONFIG SET maxmemory} {
```

Comment:
```
# https://github.com/redis/redis/runs/4630759823?check_suite_focus=true#step:6:400
[err]: module propagation with timer and CONFIG SET maxmemory in tests/unit/moduleapi/propagate.tcl
Expected '' to match 'incr notifications' (context: type source line 822 file /home/runner/work/redis/redis/tests/test_helper.tcl cmd {assert_match [lindex $patterns $j] [read_from_replication_stream $s]} proc ::assert_replication_stream level 1)
```

The test failed once

Reply:
happened again: https://github.com/redis/redis/runs/4638323673?check_suite_focus=true

Reply:
https://github.com/redis/redis/pull/10012